### ISSUE TITLE
rospy publishes /clock very slowly

### ISSUE THREAD

**Issue by trainman419** [ros/ros_comm#432]
When publishing the /clock topic, rospy publishes at a rate of around 10Hz, and this seems to get worse over time.

I can reproduce this with the following node, even with the `/use_sim_time` parameter set to false.

```
#!/usr/bin/env python

import rospy
from rosgraph_msgs.msg import Clock
from std_msgs.msg import String

rospy.init_node('tmp')
pub = rospy.Publisher('/clock', Clock, queue_size=10)
print "node initialized"

while not rospy.is_shutdown():
    t = rospy.Time.now()
    msg = Clock()
    msg.clock = t
    print t.to_sec()
    pub.publish(msg)
```

**Comment by trainman419** [comment]
This happens even if I pass the `disable_rostime` option to `rospy.init_node()`

This doesn't happen if I omit the `queue_size` option to the publisher, or if I explicitly set `queue_size=None`

**Comment by dirk-thomas** [comment]
I ran you script and it publishes messages with roughly 9kHz - I can't see any slowdown.

Can you please provide more information what you run, what the actual output is and what you expect to see?

**Comment by trainman419** [comment]
I expect to see a high publish rate; 9kHz seems reasonable.

Instead, I get this output:

```
1402086061.05
1402086061.26
1402086061.38
1402086061.54
1402086061.76
1402086061.97
1402086062.18
1402086062.35
1402086062.41
1402086062.62
1402086062.69
1402086062.9
1402086063.01
1402086063.13
1402086063.34
1402086063.56
1402086063.67
1402086063.83
1402086063.95
1402086064.01
```

I restarted my ROS master and tried to reproduce this, and it only appears to happen when there are several listeners on the /clock topic.

In particular, I can reproduce this if I set `/use_sim_time` to true, pass `disable_rostime=True` to my init_node, and run ten instances of the roscpp_listener:

```
for((i=0;i<10;i++)) ; do rosrun roscpp_tutorials listener __name:=listener_$i & done
```

**Comment by dirk-thomas** [comment]
I don't know what you mean with "disable_rostime option to rospy.init_node()" and with 10 subscribers it still publishes with that high frequence for me.

Please provide a reproducible example.

**Comment by trainman419** [comment]
Try this:

```
#!/usr/bin/env python

import rospy
from rosgraph_msgs.msg import Clock
from std_msgs.msg import String

rospy.init_node('tmp', disable_rostime=True)
pub = rospy.Publisher('/clock', Clock, queue_size=10)
print "node initialized"

while not rospy.is_shutdown():
    t = rospy.Time.now()
    msg = Clock()
    msg.clock = t
    print t.to_sec()
    pub.publish(msg)
```

**Comment by dirk-thomas** [comment]
Still working for me. Publishing rate goes down from 9kHz to about 1kHz when 10 subscribers are listening but that is the expected performance in Python.

**Comment by trainman419** [comment]
I can reproduce with the tmp.launch file in the following package:

```
git clone https://gist.github.com/c1037ea1ba8c6e683721.git tmp
```

After running for 2 or 3 seconds it slows to about 10 Hz.

**Comment by trainman419** [comment]
I'm using Hydro ros_comm 1.10.2-0precise-20140304-0128-+0000 on Precise.

Is it possible that this is the same as https://github.com/ros/ros_comm/issues/373 , and that it's fixed in Indigo?

**Comment by dirk-thomas** [comment]
Yes, that is very likely. Since you pointed to _Indigo_ code before I assumed you are actually running _Indigo_. Please confirm that it works for you with Indigo or a checkout of `ros_comm` from the _hydro-devel_ branch and close this.

**Comment by trainman419** [comment]
Confirmed; this issue is present in hydro-devel 1.10.2 and is fixed in 1.10.3
